javascript/new-src/src/factories/ola.js

```diff
@@ -22,6 +22,19 @@
 ola.factory('$ola', ['$http', '$window', 'OLA',
   function($http, $window, OLA) {
     'use strict';
+    var highestChannelNumberUsed = 0;
+
+    var updateHighestChannelNumberUsed = function(dmx) {
+      for (var channel = dmx.length; channel > highestChannelNumberUsed;
+           channel--) {
+
+        if ((dmx[channel - 1] > OLA.MIN_CHANNEL_VALUE) &&
+            (highestChannelNumberUsed < channel)) {
```

Comment:
If it did say (which as below I don't think it needs to, swapping the two variables around so it matches the for loop would be less confusing).

Reply:
When using `max` only the `> OLA.MIN_CHANNEL_VALUE` part is needed. `highestChannelNumberUsed < channel` will be removed.